src/server/game/Spells/SpellDefines.h

```diff
@@ -462,6 +463,7 @@ struct TC_GAME_API CastSpellExtraArgs
     Difficulty CastDifficulty = Difficulty(0);
     ObjectGuid OriginalCastId = ObjectGuid::Empty;
     Optional<int32> OriginalCastItemLevel;
+    Milliseconds Delay;
```

Comment:
Not sure if I should set a default value of 0ms here or convert it to a <Optional> value. The later won't allow us to use args.Delay to be passed for the AddEventAtOffset's Milliseconds offset.

Reply:
Optional is used when "no value" has a different meaning than "default value" (which would be 0ms in this case)